src/strategies/spidStrategy.ts

```diff
@@ -2,21 +2,29 @@
  * Builds and configure a Passport strategy to authenticate the proxy to the
  * different SPID IDPs.
  */
-
+// tslint:disable: no-object-mutation
 import * as SpidStrategy from "spid-passport";
 import { SpidUser } from "../types/user";
+import {
+  fetchIdpMetadata,
+  IDPMetadataOptions,
+  parseIdpMetadata
+} from "../utils/idpLoader";
 import { log } from "../utils/logger";
 
-const spidStrategy = (
+const spidStrategy = async (
   samlKey: string,
   samlCallbackUrl: string,
   samlIssuer: string,
   samlAcceptedClockSkewMs: number,
   samlAttributeConsumingServiceIndex: number,
   spidAutologin: string,
-  spidTestEnvUrl: string
+  spidTestEnvUrl: string,
+  IDPMetadataUrl: string
+  // tslint:disable-next-line: parameters-max-number
 ) => {
-  const options = {
+  // tslint:disable-next-line: no-any
+  const options: { idp: { [key: string]: IDPMetadataOptions }; sp: any } = {
     idp: {
       arubaid: {
```

Comment:
I guess these needs to be removed as they are now extracted from the XML?

Reply:
fix on commit 1327add